addons/mail/models/discuss/discuss_call_history.py

```diff
@@ -0,0 +1,27 @@
+from odoo import api, models, fields
+from odoo.addons.mail.tools.discuss import Store
+
+
+class DiscussCallHistory(models.Model):
+    _name = "discuss.call.history"
+    _description = "Keep the call history"
+
+    channel_id = fields.Many2one("discuss.channel", index=True, ondelete="cascade")
+    duration = fields.Float(compute="_compute_duration", store=True)
+    end_dt = fields.Datetime()
+    start_call_message_id = fields.Many2one("mail.message", index="btree_not_null")
+
+    _message_id_unique_constraint = models.Constraint(
+        "UNIQUE (start_call_message_id)", "Messages can only be linked to one call history"
+    )
+
+    def _to_store_defaults(self):
+        return ["create_date", "duration", "end_dt", Store.One("start_call_message_id", ["id"])]
+
+    @api.depends("create_date", "end_dt")
+    def _compute_duration(self):
+        for history in self:
+            if not history.end_dt:
+                history.duration = None
+            else:
+                history.duration = (history.end_dt - history.create_date).total_seconds() / 3600
```

Comment:
I don't like using the magic fields (`create_date` in this case) for business purpose, I'd rather if we also have a `start_dt`.

Reply:
They should contain the same value. This feels wasteful.